src/commands/cmd_bloom_filter.cc

```diff
@@ -147,6 +151,103 @@ class CommandBFMAdd : public Commander {
   std::vector<Slice> items_;
 };
 
+class CommandBFInsert : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    CommandParser parser(args, 2);
+    bool is_nonscaling = false;
+    bool has_expansion = false;
+    while (parser.Good()) {
+      if (parser.EatEqICase("capacity")) {
+        auto parse_capacity = parser.TakeInt<uint32_t>();
+        if (!parse_capacity.IsOK()) {
+          return {Status::RedisParseErr, "Bad capacity"};
+        }
+        insert_options_.capacity = parse_capacity.GetValue();
+        if (insert_options_.capacity <= 0) {
+          return {Status::RedisParseErr, "capacity should be larger than 0"};
+        }
+      } else if (parser.EatEqICase("error")) {
+        auto parse_error_rate = parser.TakeFloat<double>();
+        if (!parse_error_rate.IsOK()) {
+          return {Status::RedisParseErr, "Bad error rate"};
+        }
+        insert_options_.error_rate = parse_error_rate.GetValue();
+        if (insert_options_.error_rate >= 1 || insert_options_.error_rate <= 0) {
+          return {Status::RedisParseErr, "error rate should be between 0 and 1"};
+        }
+      } else if (parser.EatEqICase("nocreate")) {
+        insert_options_.auto_create = false;
+      } else if (parser.EatEqICase("nonscaling")) {
+        is_nonscaling = true;
+        insert_options_.expansion = 0;
+      } else if (parser.EatEqICase("expansion")) {
+        has_expansion = true;
+        auto parse_expansion = parser.TakeInt<uint16_t>();
+        if (!parse_expansion.IsOK()) {
+          return {Status::RedisParseErr, "Bad expansion"};
+        }
+        insert_options_.expansion = parse_expansion.GetValue();
+        if (insert_options_.expansion < 1) {
+          return {Status::RedisParseErr, "expansion should be greater or equal to 1"};
+        }
+      } else if (parser.EatEqICase("items")) {
+        break;
+      } else {
+        return {Status::RedisParseErr, errInvalidSyntax};
+      }
+    }
+
+    if (is_nonscaling && has_expansion) {
+      return {Status::RedisParseErr, "nonscaling filters cannot expand"};
+    }
+
+    auto items_begin =
+        find_if(args.begin(), args.end(), [](std::string_view lhs) { return lhs == "items" || lhs == "ITEMS"; });
+
+    if (items_begin == args.end()) {
+      return {Status::RedisParseErr, errInvalidSyntax};
+    }
+    items_.reserve(args_.end() - items_begin);
+    for (auto item_iter = items_begin + 1; item_iter < args.end(); ++item_iter) {
+      items_.emplace_back(*item_iter);
+    }
```

Comment:
```suggestion
      } else if (parser.EatEqICase("items")) {
        has_items = true;
        break;
      } else {
        return {Status::RedisParseErr, errInvalidSyntax};
      }
    }

    if (is_nonscaling && has_expansion) {
      return {Status::RedisParseErr, "nonscaling filters cannot expand"};
    }

    if (not has_items) {
      return {Status::RedisParseErr, errInvalidSyntax};
    }
    while (parser.Good()) {
      items_.emplace_back(GET_OR_RET(parser.TakeStr()));
    }
```

Reply:
If you want a most intuitive way, you can let args_ typed vector<string>.

Another way is to use RawTake().

Reply:
Sure, I think vector<string> would be more intutive